src/LanguageServer/Protocol/Handler/CodeActions/CodeActionResolveHelper.cs

```diff
@@ -184,6 +184,15 @@ await AddTextDocumentEditsAsync(
                     newSolution.GetDocument,
                     solution.GetDocument).ConfigureAwait(false);
 
+                // Changed source generated documents
+                if (newSolution.CompilationState.FrozenSourceGeneratedDocumentStates is not null)
+                {
+                    await AddTextDocumentEditsAsync(
+                        changes.GetChangedFrozenSourceGeneratedDocuments(),
+                        newSolution.GetRequiredSourceGeneratedDocumentForAlreadyGeneratedId,
+                        solution.GetRequiredSourceGeneratedDocumentForAlreadyGeneratedId).ConfigureAwait(false);
+                }
```

Comment:
not  sure i undersatnd this at all.  likely needs docs.  is this covered by tests? :)

Reply:
Covered by tests in the Razor repo, yes. I added a comment